### ISSUE TITLE
acoording the training scripts to train, but I can not get the results of the paper

### ISSUE THREAD

**Issue by lantianyueming** [Atten4Vis/MS-DETR#23]
the training scripts are as follows:
GPUS_PER_NODE=8 ./tools/run_dist_launch.sh 8 python -u main.py \
   --output_dir $EXP_DIR \
   --with_box_refine \
   --two_stage \
   --dim_feedforward 2048 \
   --epochs 12 \
   --lr_drop 11 \
   --coco_path=$coco_path \
   --num_queries 300 \
   --use_ms_detr \
   --use_aux_ffn \
   --cls_loss_coef 1 \
   --o2m_cls_loss_coef 2

supplement: use one RTX 4090 to train, not using distributed.

the training log and the results of the process of the training are added as an attachment.

[log.txt](https://github.com/user-attachments/files/18620464/log.txt)
[traing_model_results.txt](https://github.com/user-attachments/files/18620465/traing_model_results.txt)

**Comment by lantianyueming** [comment]
Dear author, the training is according to train_ms_detr_300.sh, could you please tell me the training setting, environment,etc in detail ? I would be very grateful to you.

**Comment by ZhaoChuyang** [comment]
Hi, the train_ms_detr_300.sh script is trained on 8 * V100 GPU. Do you modify the batch size per GPU. I did not run model on a single GPU and do not know whether the changes in global batch size will affect the final results.

Here is the training log of ms-detr run with train_ms_detr_300.sh, it seems your loss, grad_norm, and the results of the first epoch are weird. Can you provide more information about your training script or run the Deformable-DETR baseline without MS-DETR to verify the influence of single-process training?

[ms-detr-300 (2).txt](https://github.com/user-attachments/files/18621011/ms-detr-300.2.txt)

**Comment by lantianyueming** [comment]
> Hi, the train_ms_detr_300.sh script is trained on 8 * V100 GPU. Do you modify the batch size per GPU. I did not run model on a single GPU and do not know whether the changes in global batch size will affect the final results.
> 
> Here is the training log of ms-detr run with train_ms_detr_300.sh, it seems your loss, grad_norm, and the results of the first epoch are weird. Can you provide more information about your training script or run the Deformable-DETR baseline without MS-DETR to verify the influence of single-process training?
> 
> [ms-detr-300 (2).txt](https://github.com/user-attachments/files/18621011/ms-detr-300.2.txt)

Hi, I really appreciate your reply very much! And I have trained the model. The training script is as follows:
python main.py  --output_dir /home/y/MS-DETR-main/trained_model --with_box_refine --two_stage --dim_feedforward 2048 --epochs 50 --lr_drop 11 --coco_path /home/y/data --num_queries 300 --use_ms_detr  --cls_loss_coef 1 --o2m_cls_loss_coef 2 --batch_size 2 --use_aux_ffn

And then the strange result shown above appeared, I do not know why, because I completely set according to the training log of the paper, except for only a single GPU, other parameter settings are the same. Thanks for your remind! I have only one RTX 4090, I will run the model on a single RTX 4090 to verify the influence of single-process training. Eventually, thank you very much!

**Comment by Sun15194** [comment]
Have you sovled this problem? We ran almost the same results with two NVIDIA 4090s as you did with a single 4090. And we don't know what's causing it while making sure there's nothing wrong with the code.

**Comment by Sun15194** [comment]
> the training scripts are as follows: GPUS_PER_NODE=8 ./tools/run_dist_launch.sh 8 python -u main.py --output_dir $EXP_DIR --with_box_refine --two_stage --dim_feedforward 2048 --epochs 12 --lr_drop 11 --coco_path=$coco_path --num_queries 300 --use_ms_detr --use_aux_ffn --cls_loss_coef 1 --o2m_cls_loss_coef 2
> 
> supplement: use one RTX 4090 to train, not using distributed.
> 
> the training log and the results of the process of the training are added as an attachment.
> 
> [log.txt](https://github.com/user-attachments/files/18620464/log.txt) [traing_model_results.txt](https://github.com/user-attachments/files/18620465/traing_model_results.txt)

Have you solved this problem? We ran almost the same results with two NVIDIA 4090s as you did with a single 4090. And we don't know what's causing it while making sure there's nothing wrong with the code.

**Comment by lantianyueming** [comment]
> > the training scripts are as follows: GPUS_PER_NODE=8 ./tools/run_dist_launch.sh 8 python -u main.py --output_dir $EXP_DIR --with_box_refine --two_stage --dim_feedforward 2048 --epochs 12 --lr_drop 11 --coco_path=$coco_path --num_queries 300 --use_ms_detr --use_aux_ffn --cls_loss_coef 1 --o2m_cls_loss_coef 2
> > supplement: use one RTX 4090 to train, not using distributed.
> > the training log and the results of the process of the training are added as an attachment.
> > [log.txt](https://github.com/user-attachments/files/18620464/log.txt) [traing_model_results.txt](https://github.com/user-attachments/files/18620465/traing_model_results.txt)
> 
> Have you solved this problem? We ran almost the same results with two NVIDIA 4090s as you did with a single 4090. And we don't know what's causing it while making sure there's nothing wrong with the code.

It can be solved as the author do in experiment, i.e. using 8 V100 with 16 batch_size totally.

**Comment by Sun15194** [comment]
Thank you for the reply. We addressed this issue by linearly scaling the learning rate. The total batch size and base learning rate should remain fixed. By linearly scaling the learning rate based on the number of GPUs and the batch size per GPU, we can achieve the same mAP as reported in the published paper.

**Comment by lantianyueming** [comment]
> Thank you for the reply. We addressed this issue by linearly scaling the learning rate. The total batch size and base learning rate should remain fixed. By linearly scaling the learning rate based on the number of GPUs and the batch size per GPU, we can achieve the same mAP as reported in the published paper.

Congratulations very much！
